### PR TITLE
Implemented dark mode

### PR DESCRIPTION
**Pull request by drsounds**
## Hey, I just made a Pull Request!

I added support for Dark mode through the window API

````js
window.matchMedia && window.matchMedia('(prefers-color-scheme: dark)').matches
````
Screenshot:

You can now switch between dark, light or dark/light depending on system setting.

This branch replaces the defunct branch feature/dark-mode-suppot

#### :heavy_check_mark: Checklist

- [ ] All tests are passing `yarn test`
- [x] Screenshots attached (for UI changes)
- [ ] Relevant documentation updated
- [ ] Prettier run on changed files
- [ ] Tests added for new functionality
- [ ] [CODEOWNERS](./CODEOWNERS) updated (for new stuff)
- [ ] Regression tests added for bug fixes

### CODE REVIEW

**Comment by drsounds** [comment]
It seems like it breaks on this code

````
Download the React DevTools for a better development experience: https://fb.me/react-devtools
Warning: React.createFactory() is deprecated and will be removed in a future major release. Consider using JSX or use React.createElement() directly instead.
The above error occurred in the <App> component:
    in App (at src/index.tsx:21)
````

**Comment by drsounds** [comment]
It seems to be something that is problem becauase of new React version. It doesn't happen on my computer. I now tried to create new branch and start over but i still get the same error

**Comment by drsounds** [comment]
Yarn start works like a charm for me on my iMac 27" macOS 10.15.4 Catalina Node 12

````
buddhalow@iMacsomlexander bs % yarn start      
yarn run v1.21.1
$ yarn build && yarn workspace example-app start
$ lerna run build
lerna notice cli v3.20.2
lerna info Executing command in 4 packages: "yarn run build"
lerna info run Ran npm script 'build' in '@backstage/cli' in 2.1s:
$ backstage-cli build-cache -- tsc --outDir dist --noEmit false --module CommonJS
[build-cache] cache hit, nothing to be done
lerna info run Ran npm script 'build' in '@backstage/core' in 1.8s:
$ backstage-cli build-cache -- tsc --outDir dist/cjs --noEmit false --module CommonJS
[build-cache] cache hit, nothing to be done
lerna info run Ran npm script 'build' in '@backstage/plugin-welcome' in 1.8s:
$ backstage-cli build-cache -- backstage-cli plugin:build
[build-cache] cache hit, nothing to be done
lerna info run Ran npm script 'build' in '@backstage/plugin-home-page' in 1.9s:
$ backstage-cli build-cache -- backstage-cli plugin:build
[build-cache] cache hit, nothing to be done
lerna success run Ran npm script 'build' in 4 packages in 5.8s:
lerna success - @backstage/cli
lerna success - @backstage/core
lerna success - @backstage/plugin-home-page
lerna success - @backstage/plugin-welcome
$ backstage-cli app:serve
Compiled successfully!

You can now view example-app in the browser.

  Local:            http://localhost:3000
  On Your Network:  http://192.168.1.203:3000

Note that the development build is not optimized.
To create a production build, use npm run build.



````

But in CLI Test / Node 12.x it breaks 

````
<--- JS stacktrace --->

==== JS stack trace =========================================

Security context: 0x16e0788008d1 <JSObject>
    0: builtin exit frame: parse(this=0x16e07881ec89 <Object map = 0x45dd1183549>,0x0121dd100119 <Very long string[7644721]>,0x16e07881ec89 <Object map = 0x45dd1183549>)

    1: /* anonymous */(aka /* anonymous */) [0x15429a0943d9] [/home/runner/work/backstage/backstage/node_modules/zombie/lib/fetch.js:260] [bytecode=0x26c01d6818e1 offset=22](this=0x1252632404b1 <undefined>)
    2: StubFrame [pc...

 1: 0x9ef190 node::Abort() [node]
 2: 0x9f13b2 node::OnFatalError(char const*, char const*) [node]
 3: 0xb5da9e v8::Utils::ReportOOMFailure(v8::internal::Isolate*, char const*, bool) [node]
 4: 0xb5de19 v8::internal::V8::FatalProcessOutOfMemory(v8::internal::Isolate*, char const*, bool) [node]
 5: 0xd0a765  [node]
 6: 0xd0adf6 v8::internal::Heap::RecomputeLimits(v8::internal::GarbageCollector) [node]
 7: 0xd1760a v8::internal::Heap::PerformGarbageCollection(v8::internal::GarbageCollector, v8::GCCallbackFlags) [node]
 8: 0xd18515 v8::internal::Heap::CollectGarbage(v8::internal::AllocationSpace, v8::internal::GarbageCollectionReason, v8::GCCallbackFlags) [node]
 9: 0xd1afcc v8::internal::Heap::AllocateRawWithRetryOrFail(int, v8::internal::AllocationType, v8::internal::AllocationOrigin, v8::internal::AllocationAlignment) [node]
10: 0xce8c8c v8::internal::Factory::NewRawOneByteString(int, v8::internal::AllocationType) [node]
11: 0xdf815b v8::internal::JsonParser<unsigned short>::MakeString(v8::internal::JsonString const&, v8::internal::Handle<v8::internal::String>) [node]
12: 0xdfa9fe v8::internal::JsonParser<unsigned short>::ParseJsonValue() [node]
13: 0xdfb1cf v8::internal::JsonParser<unsigned short>::ParseJson() [node]
14: 0xc0f035 v8::internal::Builtin_JsonParse(int, unsigned long*, v8::internal::Isolate*) [node]
15: 0x13a72b9  [node]
/home/runner/work/_temp/bb0995c5-dbd7-4952-ba2c-97738075befc.sh: line 2:  3424 Aborted                 (core dumped) node scripts/cli-e2e-test.js
##[error]Process completed with exit code 134.
````

But I can get the app to work on my iMac

**Comment by Rugvip** [comment]
#442 will help out debugging this issue.

In this case the actual error you're getting is `window.matchMedia is not a function`, so you just want to do a feature check for that before you use it. I'd say just assume light theme if it isn't present.

**Comment by drsounds** [comment]
window.matchMedia is not the problem with CLI Test, it seems that there is some strange error about React.createFactory is deprecated...

File: packages/app/src/App.tsx [hunk 1]
```diff
@@ -48,17 +53,42 @@ const AppComponent = app.build();
...
+  let backstageTheme = BackstageThemeLight;
+  switch (theme) {
+    case 'light':
+      backstageTheme = BackstageThemeLight;
+      break;
+    case 'dark':
+      backstageTheme = BackstageThemeDark;
+      break;
+    default:
+      backstageTheme = window.matchMedia('(prefers-color-scheme: dark)').matches
```
Comment (Rugvip) [inline packages/app/src/App.tsx:69]: Needs a check here as well

Comment (drsounds) [reply packages/app/src/App.tsx:73]: I've now fixed this.

**Review by nikek — APPROVED** [review]

**Review by Rugvip — APPROVED** [review]
👍 , but let's add an issue for merging the themes back together

**Comment by drsounds** [comment]
Great! What do you mean merging the themes back together? I remember you in a previous review told me to split up the themesinto two

**Comment by Rugvip** [comment]
Are you referring to this comment? https://github.com/spotify/backstage/pull/398#issuecomment-605054213

I did say two themes, but in the same file. And they would naturally share most of the definitions. That's what the issue is about anyway, feel free to pick it up if you want! 😄 

At this stage we lean towards merging PRs early and iterating in master, as there are too many PRs changing the same parts of the code to have long-lived PRs.

**Comment by drsounds** [comment]
I tried to add that email but it didnt help

**Pull request merged**